src/main/java/com/example/cookbook/recipe/RecipeService.java

```diff
@@ -0,0 +1,60 @@
+package com.example.cookbook.recipe;
+
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+@Service
+public class RecipeService {
+    private RecipeRepository recipeRepository;
+    private RecipeDtoMapper recipeDtoMapper;
+
+    public RecipeService(RecipeRepository recipeRepository, RecipeDtoMapper recipeDtoMapper) {
+        this.recipeRepository = recipeRepository;
+        this.recipeDtoMapper = recipeDtoMapper;
+    }
+
+    RecipeDto getRecipeById(Long id) {
+        return recipeRepository.findById(id)
+                .map(recipeDtoMapper::map).get();
+    }
+
+    public void addLike(Long id) {
+        RecipeDto recipeDto = getRecipeById(id);
+        Recipe recipe = recipeRepository.findById(recipeDto.getId()).get(); //czy nie jest zla praktyka od razu pominac ta linie i odwolac sie do recipe.liked?
+        recipe.setLikedRecipe(recipe.getLikedRecipe() + 1);
```

Comment:
Jest ok, ale jak używamy Optionala, to dobrze jest wyróżnić dwie sytuacje, czyli że obiekt o danym id istnieje, albo nie.
```
Recipe recipe = recipeRepository.findById(recipeDto.getId()).orElseThrow();
recipe.setLikedRecipe(recipe.getLikedRecipe() + 1);
```